### PR TITLE
nrf_modem: fix binary linking for upmerge 25-03-2024

### PR DESCRIPTION
**Pull request by MirkoCovizzi**
This commit updates the cmake configuration file to
recognize the new SOC names defined with HWMv2.

### CODE REVIEW

File: nrf_modem/CMakeLists.txt [hunk 1]
```diff
@@ -6,10 +6,15 @@
...
+  if(NOT ${CONFIG_SOC} MATCHES "^nrf9160$" AND NOT ${CONFIG_SOC} MATCHES "^nrf91[356]1$")
+    message(FATAL_ERROR "Unknown SOC. Expected (nrf9160, nrf9131, nrf9151, nrf9161), "
+                        "got '${CONFIG_SOC}'.")
+  endif()
+
+  string(REPLACE "nrf" "nRF" arch_soc_dir ${CONFIG_SOC})
 
-  if(NOT arch_soc_dir MATCHES "nRF9160")
-  string(REGEX REPLACE "nRF91[0-9]*" "nRF9120" arch_soc_dir ${arch_soc_dir})
+  if(NOT arch_soc_dir MATCHES "^nRF9160$")
```
Comment (eivindj-nordic) [inline nrf_modem/CMakeLists.txt:16]:
```suggestion
  if(${CONFIG_SOC} MATCHES "^nrf91[356]1$")
```

Comment (MirkoCovizzi) [reply nrf_modem/CMakeLists.txt:16]: Actually, I think we can completely remove the if, because the substitution happens only if the regex matches.

**Review by eivindj-nordic — APPROVED** [review]
Looks good to me, though it has to be linked or cherry-picked in the upmerge.

**Comment by MirkoCovizzi** [comment]
> Looks good to me, though it has to be linked or cherry-picked in the upmerge.

I'm aware, I've already notified the upmerge maintainer.

**Review by anhmolt — APPROVED** [review]

**Pull request closed without merging**